charms/metallb-speaker/src/utils.py

```diff
@@ -169,11 +176,14 @@ def delete_namespaced_role_with_api(name, namespace):
             api_instance.delete_namespaced_role(
                 name=name, namespace=namespace, body=body, pretty=True
             )
-        except ApiException:
+        except ApiException as err:
             logging.exception(
                 "Exception when calling RbacAuthorizationV1Api"
                 "->delete_namespaced_role."
             )
+            if err.status != 409:
+                # Hook error except for 409 (AlreadyExists) errors
+                sys.exit(1)
```

Comment:
This will raise a [SystemExit](https://docs.python.org/3/library/exceptions.html#SystemExit) exception that will bypass `except Exception` clauses because it does not inherit from Exception. It could be very disruptive to any code that's calling this.

Can we `raise` instead? I think we should give callers a chance to handle whatever's going wrong here.

```suggestion
                raise
```

Reply:
Hmm, I see you followed an existing pattern in the charm, here. Ok. I'll leave this as a non-blocking nit.